lib/rpmsg/rpmsg_virtio.c

```diff
@@ -369,6 +375,10 @@ static int rpmsg_virtio_send_offchannel_nocopy(struct rpmsg_device *rdev,
 				      &rp_hdr, sizeof(rp_hdr));
 	RPMSG_ASSERT(status == sizeof(rp_hdr), "failed to write header\r\n");
 
+#ifdef VIRTIO_CACHED_BUFFERS
+	metal_cache_flush((void *)data, len);
```

Comment:
size to flush is len ( payload) + sizeof(hdr)
But it would be more symmetrical to do this in rpmsg_virtio_enqueue_buffer.

Reply:
yeah, that seems better